lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -346,6 +368,184 @@ static LogicalResult verifyCircuitOp(CircuitOp circuit) {
     }
   }
 
+  // Verify the NonLocalAnchor.
+  // 1. Iterate over the namepath.
+  // 2. The namepath should be a valid instance path, specified either on a
+  // module or a declaration inside a module.
+  // 3. Each element in the namepath is an InnerRefAttr except possibly the
+  // last element.
+  // 4. Make sure that the InnerRefAttr is legal, by verifying the module name
+  // and the corresponding inner_sym on the instance.
+  // 5. Make sure that the instance path is legal, by verifying the sequence of
+  // instance and the expected module occurs as the next element in the path.
```

Comment:
Not sure if this one is working properly, the following example does not cause a verification error:
```mlir
module  {
  firrtl.circuit "Top"   {
   // Legal nla would be:
  //firrtl.nla @nla [#hw.innerNameRef<@Top::@mid>, #hw.innerNameRef<@Mid::@leaf>, #hw.innerNameRef<@Leaf::@w>]
    firrtl.nla @nla [#hw.innerNameRef<@Top::@mid>, #hw.innerNameRef<@Leaf::@w>]
    firrtl.module @Leaf() {
      %w = firrtl.wire sym @w  {annotations = [{circt.nonlocal = @nla, class = "fake1"}]} : !firrtl.uint<3>
    }
    firrtl.module @Middle() {
      firrtl.instance leaf sym @leaf  {annotations = [{circt.nonlocal = @nla, class = "circt.nonlocal"}]} @Leaf()
    }
    firrtl.module @Top() {
      firrtl.instance mid sym @mid  {annotations = [{circt.nonlocal = @nla, class = "circt.nonlocal"}]} @Middle()
    }
  }
}
```

Reply:
Fixed the issue, and added as a lit test.